node/src/blockchain/blockchain_interface_utils.rs

```diff
@@ -0,0 +1,1412 @@
+            payable.balance_wei.separate_with_commas(),
+            payable.wallet,
+            pending_nonce
+        );
+
+        let hash_and_amount = sign_and_append_payment(
+            chain,
+            web3_batch.clone(),
+            consuming_wallet.clone(),
+            pending_nonce,
+            gas_price,
+            payable,
+        );
+
+        pending_nonce = advance_used_nonce(pending_nonce);
+        hash_and_amount_list.push(hash_and_amount);
+    });
+    hash_and_amount_list
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn send_payables_within_batch(
+    logger: Logger,
+    chain: Chain,
+    web3_batch: Web3<Batch<Http>>,
+    consuming_wallet: Wallet,
+    gas_price_in_gwei: u64,
+    pending_nonce: U256,
+    new_fingerprints_recipient: Recipient<PendingPayableFingerprintSeeds>,
+    accounts: Vec<PayableAccount>,
+) -> Box<dyn Future<Item = Vec<ProcessedPayableFallible>, Error = PayableTransactionError> + 'static>
+{
+    debug!(
+            logger,
+            "Common attributes of payables to be transacted: sender wallet: {}, contract: {:?}, chain_id: {}, gas_price: {}",
+            consuming_wallet,
+            chain.rec().contract,
+            chain.rec().num_chain_id,
+            gas_price_in_gwei
+        );
+
+    let hashes_and_paid_amounts = sign_and_append_multiple_payments(
+        logger.clone(),
+        chain,
+        web3_batch.clone(),
+        consuming_wallet,
+        gas_price_in_gwei,
+        pending_nonce,
+        accounts.clone(),
+    );
+
+    let timestamp = SystemTime::now();
+    let hashes_and_paid_amounts_error = hashes_and_paid_amounts.clone();
+    let hashes_and_paid_amounts_ok = hashes_and_paid_amounts.clone();
+
+    // TODO: We are sending hashes_and_paid_amounts to the Accountant even if the payments fail.
+    new_fingerprints_recipient
+        .try_send(PendingPayableFingerprintSeeds {
+            batch_wide_timestamp: timestamp,
+            hashes_and_balances: hashes_and_paid_amounts,
+        })
+        .expect("Accountant is dead");
+
+    info!(
+        logger,
+        "{}",
+        transmission_log(chain, &accounts, gas_price_in_gwei)
+    );
+
+    return Box::new(
+        web3_batch
+            .transport()
+            .submit_batch()
+            .map_err(|e| error_with_hashes(e, hashes_and_paid_amounts_error))
+            .and_then(move |batch_response| {
+                Ok(merged_output_data(
+                    batch_response,
+                    hashes_and_paid_amounts_ok,
+                    accounts,
+                ))
+            }),
+    );
+}
+
+pub fn calculate_fallback_start_block_number(start_block_number: u64, max_block_count: u64) -> u64 {
+    if max_block_count == u64::MAX {
+        start_block_number + 1u64
+    } else {
+        start_block_number + max_block_count
+    }
+}
+
+pub fn convert_wei_to_gwei(wei: U256) -> u64 {
+    (wei / U256::from(GWEI_UNIT)).as_u64() + 1
```

Comment:
@bertllll 
This is due to the fact that we are rounding off. Its protects the case were values are smaller that 1000000000.
So currently we are added +1 Gwei when getting the gas price.

We did this as a temporary measure. should be fixed when we add full support for wei

This came about because we are now getting gas_price from the RPC instead of persistent_config.
The RPC return gas_price in wei, but our code base only supports Gwei.

Reply:
Yeah, after you explained me, I have understanding for your way. Still, I think we can afford getting down to the replacement of gwei by wei. Fragmenting the work into cards would be unnecessary hassle. I'm talking about the trade-offs and I see them speak for doing it now. Each approach suits one time and doesn't some other.